images/linux/scripts/installers/runner-package.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -e
+################################################################################
+##  File:  runner-package.sh
+##  Desc:  Downloads and Installs runner package
+################################################################################
+
+# Source the helpers for use with the script
+source $HELPER_SCRIPTS/install.sh
+
+DOWNLOAD_URL=$(get_github_package_download_url "actions/runner" "test(\"actions-runner-linux-x64-[0-9].[0-9]{3}.[0-9].tar.gz\")" "latest")
+
+FILE_NAME="${DOWNLOAD_URL##*/}"
+
+sudo mkdir -p /opt/runner-cache
+
+download_with_retries "${DOWNLOAD_URL}" "/opt/runner-cache" "${FILE_NAME}"
```

Comment:
you can not save directly to `/opt/runner-cache` as it is write-protected, so you need to save it to `/tmp` and only then `sudo mv` it from `/tmp` to target destination

Reply:
I think the current implementation should work because the full script `runner-package.sh` is run from `root` user: https://github.com/actions/runner-images/blob/433178adf5d64cca48a8bf96525b38385e3b7682/images/linux/ubuntu2204.pkr.hcl#L281

Reply:
@mikhailkoliada could you help me understand why `/opt/runner-cache` is write-protected? 😃 and if it is still a blocker if, as Maxim mentioned above, we are running image generation as root